tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,399 @@
+
+        :id: b3de6f96-7c39-4c44-b91c-a6d141f5dd6a
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having single key and multiple values
+                separated with comma
+
+        :expectedresults: The parameter with multiple values should be saved
+            in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_multiple_names(self):
+        """Subnet parameters can be created with multiple names with valid
+        separators
+
+        :id: d1e2d75a-a1e8-4767-93f1-0bb1b75e10a0
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by valid separators(e.g fwd slash) and value
+
+        :expectedresults: The parameter with multiple names separated by valid
+            separators should be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_with_parameter_and_invalid_separator(self):
+        """Subnet parameters can not be created with multiple names with
+        invalid separators
+
+        :id: 08d10b75-a0db-4a11-a915-965a2a207d16
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by invalid separators(e.g spaces) and value
+
+        :expectedresults: The parameter with multiple names separated by
+            invalid separators should not be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_with_duplicated_parameters(self):
+        """Duplicate parameters cannot be created in Subnet
+
+        :id: aa69bdcc-e833-41e4-8f72-7139bdd64daa
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create Multiple parameters having duplicate key names
+
+        :expectedresults: The subnet parameters should not be created with
+            duplicate names
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier3
+    def test_positive_inherit_subnet_parmeters_in_host(self):
+        """Host inherits parameters from subnet
+
+        :id: e372a594-f758-40ef-95ec-867937e44b63
+
+        :steps:
+
+            1. Create valid subnet with a valid parameter
+            2. Create host with above subnet
+            3. Assign hosts primary interface with subnet
+            4. List inherited subnet parameters in above host
+
+        :expectedresults:
+
+            1. The parameters from subnet should be displayed in
+                host parameters
+            2. The parameters from subnet should be displayed in
+                host enc output
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier3
+    def test_negative_inherit_subnet_parmeters_in_host(self):
+        """Host does not inherits parameters from subnet for non primary
```

Comment:
It is more `Host does not inherits parameters from subnet for interface that was unset from host`, because `non primary` means `secondary` and it is like you have five network interface, but only third network interface in the list has parameters. By the way, are you going to see parameters from non primary interface?